src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/component/CommunityAPIClient.kt

```diff
@@ -51,22 +58,51 @@ class CommunityAPIClient(
       .bodyToMono(responseBodyClass)
       .onErrorMap { e ->
         handleResponse(e, "")
-        e
       }
       .block()
   }
 
-  fun handleResponse(e: Throwable, requestBody: Any): String {
+  fun handleResponse(e: Throwable, requestBody: Any): DownstreamApiCallError {
     val responseBodyAsString = when (e) {
       is BadRequest -> e.responseBodyAsString
       else -> e.localizedMessage
     }
-    logger.error(
-      "Call to community api failed",
-      e,
-      StructuredArguments.kv("req.body", requestBody),
-      StructuredArguments.kv("res.body", responseBodyAsString)
-    )
-    return responseBodyAsString.let { it } ?: "No response body message"
+
+    val statusCode = when (e) {
+      is WebClientResponseException -> e.statusCode
+      else -> INTERNAL_SERVER_ERROR
+    }
+
+    val causeMessage = userMessageOrDeveloperMessageOrResponseBodyInThatOrder(responseBodyAsString)
+    val error = DownstreamApiCallError(statusCode, causeMessage, responseBodyAsString, e)
```

Comment:
We will want to know which API failed during live ops. Can we establish the context by either naming this class `CommunityAPICallError` or adding a new field saying `source = "community-api"`?

Reply:
Renamed to CommunityAPICallError